src/directives/formBuilderDnd.js

```diff
@@ -31,7 +31,7 @@ module.exports = [
 
     $scope.addComponent = function(component, index) {
       // Only edit immediately for components that are not resource comps.
-      if (component.isNew && (!component.key || (component.key.indexOf('.') === -1))) {
+      if (component.isNew && !component.isNonEditable && (!component.key || (component.key.indexOf('.') === -1))) {
```

Comment:
I don't like having negatives in variable names. I understand why you did this since you want reverse compatibility, but can we call this variable something like ```isStatic```, which I suppose could be interpreted as "Configurations are static". I am open to suggestions.

Thanks for your contributions.